python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,2012 @@
+            if get_visit_id_method is not None:
+                return get_visit_id_method()
+            else:
+                return exp_id
+        except Exception:
+            return -1
+
+    def _determine_defocal_type(self, exposure: afwImage.Exposure) -> str:
+        """Determine defocal type (intra/extra) from exposure metadata.
+
+        Args:
+            exposure: The exposure to analyze
+
+        Returns:
+            String indicating defocal type: "intra" or "extra"
+
+        Raises:
+            ValueError: If defocal type cannot be determined
+        """
+        # First, try to get DFC_TYPE from exposure metadata
+        try:
+            metadata = exposure.getMetadata()
+            if metadata is not None and "DFC_TYPE" in metadata.names():
+                defocal_type = metadata.get("DFC_TYPE")
+                self.log.info("Found DFC_TYPE in exposure metadata: '%s'", defocal_type)
+                return defocal_type
+        except Exception:
+            pass
+
+        # Fallback: try to determine from focus Z value if available
+        try:
+            focus_z = exposure.visitInfo.focusZ
+            # For LSSTCam, positive focusZ is typically extra-focal
+            # This is a heuristic - may need adjustment based on actual data
+            if focus_z > 0:
+                defocal_type = "extra"
+            else:
+                defocal_type = "intra"
+            self.log.info("Determined defocal type from focusZ=%.3f: '%s'", focus_z, defocal_type)
+            return defocal_type
+        except (AttributeError, NameError):
+            # Cannot determine defocal type - this is a critical error
+            raise ValueError(
+                "Cannot determine defocal type from exposure metadata. "
+                "Neither DFC_TYPE metadata nor focusZ value is available. "
+                "This is required for proper wavefront analysis. "
+                "Please ensure exposure has valid metadata."
+            )
+
+    def _get_exposure_metadata(self, exposure: afwImage.Exposure) -> dict[str, float | int]:
+        """Extract metadata from exposure as a dictionary
+           with safe fallback values.
+
+        Args:
+            exposure: The exposure object to extract metadata from
+
+        Returns:
+            Dictionary containing metadata with safe fallback values
+        """
+        metadata: dict[str, float | int] = {}
+
+        # Get visitInfo safely - this is the correct way to access angles
+        visit_info = getattr(exposure, 'visitInfo', None)
+        if visit_info is None:
+            self.log.warning("No visitInfo available in exposure, using fallback values")
+            return metadata
+
+        # Visit ID
+        metadata['visit_id'] = getattr(visit_info, 'id', -1)
+
+        # Extract boresight angles from visitInfo (the correct approach)
+        try:
+            boresight_rot_angle = getattr(visit_info, 'boresightRotAngle', None)
```

Comment:
You have an if and else here that could be removed if you wrote 0.0 instead of None